src/main/java/com/github/princesslana/eriscasper/util/AuditLogEventType.java

```diff
@@ -0,0 +1,61 @@
+package com.github.princesslana.eriscasper.util;
+
+import java.util.Optional;
+
+public enum AuditLogEventType {
+  GUILD_UPDATE(1),
+  CHANNEL_CREATE(10),
+  CHANNEL_UPDATE(11),
+  CHANNEL_DELETE(12),
+  CHANNEL_OVERWRITE_CREATE(13),
+  CHANNEL_OVERWRITE_UPDATE(14),
+  CHANNEL_OVERWRITE_DELETE(15),
+  MEMBER_KICK(20),
+  MEMBER_PRUNE(21),
+  MEMBER_BAN_ADD(22),
+  MEMBER_BAN_REMOVE(23),
+  MEMBER_UPDATE(24),
+  MEMBER_ROLE_UPDATE(25),
+  ROLE_CREATE(30),
+  ROLE_UPDATE(31),
+  ROLE_DELETE(32),
+  INVITE_CREATE(40),
+  INVITE_UPDATE(41),
+  INVITE_DELETE(42),
+  WEBHOOK_CREATE(50),
+  WEBHOOK_UPDATE(51),
+  WEBHOOK_DELETE(52),
+  EMOJI_CREATE(60),
+  EMOJI_UPDATE(61),
+  EMOJI_DELETE(62),
+  MESSAGE_DELETE(72);
+
+  private int code;
+
+  AuditLogEventType(int code) {
+    this.code = code;
+  }
+
+  public int getCode() {
+    return code;
+  }
+
+  private static final AuditLogEventType[] types = new AuditLogEventType[72];
```

Comment:
This looks like premature optimization to me.

Reply:
It's for end users to easily be able to find out what code corresponds to the audit log they're trying to look at. It's done the same way with OP CODE

Reply:
AuditLogs tend to contain a very large sum of them, looping through them all will be slower than just grabbing the cached enum. It takes a very small amount of memory as well so why not prefer speed over memory.

Reply:
If we're going to be doing this array, then we need the `of` method to have some changes.

It shouldn't hard code `72`, we can use `types.length`, we should handle negative ints, and we should handle the case where an int is passed in that we don't have an entry for. To be consistent with OpCode an invalid value should result in an Exception.

Reply:
Gotcha, will fix